portfolio/src/main/webapp/script.js

```diff
@@ -13,23 +17,12 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
- */
-// function addRandomGreeting() {
-//   const greetings =
-//       ["What we know is a drop and what we don't know is an ocean.", "Beets, Bears, Battlestar Galactica", "To live is to risk it all.", "People don't change. They become more of who they are."];
-
-//   // Pick a random greeting.
-//   const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-//   // Add it to the page.
-//   console.log("this is a function for random quotes");
-//   const greetingContainer = document.getElementById('greeting-container');
-//   greetingContainer.innerText = greeting;
-// }
-
-function addProjectDetail(buttonId) {
-  const description =
+  * Shows details of a specified project.
+  * @param {HTMLElement.itemId} buttonId The ID of the button that was clicked to invoke the function.
+  * @return {void}
+  */
+function showProjectDetail(buttonId) {
```

Comment:
A better approach would be to use the event object received from the function call and then use event.targetElement to access the target element. There is no need to pass the id.

```
function showProjectDetail(event){
  // use event.targetElement here.
}
```

Reply:
You can then use the element.parentElement.append() to add a child div block. Currently the div is rendered without any containing text. It is better we add them dynamically.

Let's say variable `x` stores your targetElement (i.e. the button), 

```
const divElement = document.createElement('div');
divElement.innerText = 'Some details';
x.parentElement.appendChild(divElement);
```